homeassistant/components/anova/sensor.py

```diff
@@ -0,0 +1,113 @@
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfTemperature, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import AnovaCoordinator
+from .entity import AnovaEntity
+from .models import AnovaData
+
+SENSOR_DESCRIPTIONS: list[SensorEntityDescription] = [
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.COOK_TIME,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:clock-outline",
+        name="Cook Time",
+    ),
+    SensorEntityDescription(key=AnovaPrecisionCookerSensor.STATE, name="State"),
+    SensorEntityDescription(key=AnovaPrecisionCookerSensor.MODE, name="Mode"),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.TARGET_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Target Temperature",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.COOK_TIME_REMAINING,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:clock-outline",
+        name="Cook Time Remaining",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.HEATER_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Heater Temperature",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.TRIAC_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Triac Temperature",
+    ),
+    SensorEntityDescription(
+        key=AnovaPrecisionCookerSensor.WATER_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+        name="Water Temperature",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Anova device."""
+    anova_data: AnovaData = hass.data[DOMAIN][entry.entry_id]
+    sensors: list[AnovaSensor] = []
+    for coordinator in anova_data.coordinators:
+        await coordinator.async_config_entry_first_refresh()
```

Comment:
This needs to happen in `__init__` since its too late to raise `ConfigEntryNotReady` as the platform 

Also you are already doing this so its duplicate work to do it here as well

Reply:
Ah - I had that realization reviewing someone else PR but didn't realize it here. Thanks!